HpToolsLauncher/Launcher.cs

```diff
@@ -507,7 +513,7 @@ IAssetRunner CreateRunner(TestStorageType runType, JavaProperties ciParams)
                         mobileinfo = _ciParams["mobileinfo"];
                     }
 
-                    runner = new FileSystemTestsRunner(validTests, timeout, pollingInterval, perScenarioTimeOutMinutes, ignoreErrorStrings, jenkinsEnvVariables, mcConnectionInfo, mobileinfo);
+                    runner = new FileSystemTestsRunner(validTests, timeout, uftRunMode, pollingInterval, perScenarioTimeOutMinutes, ignoreErrorStrings, jenkinsEnvVariables, mcConnectionInfo, mobileinfo);
```

Comment:
There is still no decoupling in the call to the constructor.
You should call the overloaded constructor - only when needed i.e. only when the "fsUftRunMode" param is supplied.
If not  then the regular constructor should be called.